satellite/satellitedb/certdb.go

```diff
@@ -21,15 +21,20 @@ type certDB struct {
 
 func (certs *certDB) SavePublicKey(ctx context.Context, nodeID storj.NodeID, publicKey crypto.PublicKey) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	_, err = certs.db.Get_CertRecord_By_Id(ctx, dbx.CertRecord_Id(nodeID.Bytes()))
-	if err == sql.ErrNoRows {
-		return certs.tryAddPublicKey(ctx, nodeID, publicKey)
+	pubbytes, err := pkcrypto.PublicKeyToPKIX(publicKey)
```

Comment:
Ensure we have the full chain not just the leaf.

Reply:
could you be more specific.. am bit lost.. what do you mean by full chain and how do I ensure that. 

All I am doing is checking if there is an entry with similar public key and if it is then return error otherwise what we did originally , to add it to the table.

Reply:
This is unresolved.

Reply:
will address this comment in https://github.com/storj/storj/pull/2704/ PR